solvation_analysis/solvation.py

```diff
@@ -6,29 +6,142 @@
 """
 
 
-def canvas(with_attribution=True):
+# atom selection functions
+import warnings
+import numpy as np
+import MDAnalysis as mda
+from MDAnalysis.analysis import distances
+import nglview as nv
+
+# all functions
+"get_atom_group, get_n_shells, \
+    get_cation_anion_shells, get_closest_n_mol, get_radial_shell, \
+    get_counts, get_pair_type, count_dicts"
+
+
+def visualize(selection):
+    mda_view = nv.show_mdanalysis(selection)
+    mda_view.add_representation('licorice', selection='Li', color='blue')
+    return mda_view.display()
+
+
+def get_atom_group(u, selection):
+    """
+    Casts an Atom, AtomGroup, Residue, or ResidueGroup to AtomGroup.
+
+    Parameters
+    ----------
+        u : Universe
+            universe that contains central species
+        selection: Atom, AtomGroup, Residue, or ResidueGroup
+
+    Returns
+    -------
+        AtomGroup
+
+    """
+    assert isinstance(selection, (mda.core.groups.Residue,
+                                  mda.core.groups.ResidueGroup,
+                                  mda.core.groups.Atom,
+                                  mda.core.groups.AtomGroup)), \
+        "central_species must be one of the preceding types"
+    if isinstance(selection, (mda.core.groups.Residue, mda.core.groups.ResidueGroup)):
+        selection = selection.atoms
+    if isinstance(selection, mda.core.groups.Atom):
```

Comment:
u.atoms[[atom.ix]] - the double brackets turn the input to getitem into a list and therefore you get a (one length) AtomGroup out